plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -156,7 +157,9 @@ public void testDeleteRowsConcurrently()
                     .collect(toImmutableList());
 
             Stream<Optional<String>> expectedRows = Streams.mapWithIndex(futures.stream(), (future, index) -> {
```

Comment:
Let's rather reduce the concurrency 

`nThreads = 3`

```
TestTable table = new TestTable(
                getQueryRunner()::execute,
                "test_concurrent_delete",
                "(col0 INTEGER, col1 INTEGER, col2 INTEGER)")) {
```

Reply:
what is "metastore replacement operation failure spiral"?

Reply:
Potentially trying to update repeatedly the metastore entry at the same time repeatedly

Reply:
how long can it reasonably take for 4 queries?  and how long in extreme case? what would be safe test timeout? 
how does the number look like with 3 queries?

Reply:
#13995  states 

```
Error:  io.trino.plugin.iceberg.TestIcebergGcsConnectorSmokeTest.testDeleteRowsConcurrently  Time elapsed: 120.014 s  <<< FAILURE!
```

so - in some peculiar cases it seems to take more than 2 minutes. i don't see any value in investing over 2 minutes to test this functionality.

Reply:
in that issue there are two failure modes
overall test timeout after 120s. last known occurrence April 2023 https://github.com/trinodb/trino/issues/13995#issuecomment-1516820401
timeout when waiuting for individual query to complete (increaeased here) "NoSuchElementException: No value present" -- last known occurence Aug 2023 https://github.com/trinodb/trino/issues/13995#issuecomment-1680350178 and today 

this is this second problem that i am trying to change

i don't feel strongly though, if  you have a better idea how to help this test fail less frequently

Reply:
ok. let's fix the second problem.

if the timeout will happen again we can reduce the concurrency.